Exit with error on failure to start server

larsks opened the pull request:
Previously, if the server failed to start (e.g., if it was unable to load
the key or certificate), serverd would remain running indefinitely because
it was waiting for a signal.

This commit modifies the code so that if ListenAndServerTLS returns it will
cause serverd to exit.

larsks committed:
Exit with error on server failure

Previously, if the server failed to start (e.g., if it was unable to load
the key or certificate), serverd would remain running indefinitely because
it was waiting for a signal.

This commit modifies the code so that if ListenAndServerTLS returns it will
cause serverd to exit.

douglasmakey commented:
Hi @larsks, thanks so much for the help! I didn't notice that.

What do you think if we move the invocation of the ListenAndServeTLS method to the main goroutine, so the listener blocks it and we use another goroutine to handle the shutdown?

example: 

```go
	go func() {
		// listen shutdown signal
		signalChan := make(chan os.Signal, 1)
		signal.Notify(signalChan, syscall.SIGINT, syscall.SIGTERM)

		<-signalChan
		log.Infof("Shutdown gracefully...")
		if err := server.Shutdown(context.Background()); err != nil {
			log.Error(err)
		}
	}()

	log.Infof("Server running in port: %s", port)
	if err := server.ListenAndServeTLS(tlscert, tlskey); err != nil {
		log.Errorf("Failed to listen and serve: %v", err)
	}
```

Then, if the server fails due to TLS, Port or any other error, the program will exit normally. We could also add a `log.Fatal`!

larsks commented:
I've updated this PR to reflect our discussion:

- Moved the signal handling into the goroutine
- Server exist causes `serverd` exit
- Updated some of the verbage in the log messages
- `serverd` exit code is nonzero on error/signal
